dedupe/convenience.py

```diff
@@ -24,55 +24,94 @@ def consoleLabel(deduper): # pragma: no cover
+            record_pair_buffer = record_pair_buffer[1:]
+            examples_buffer = examples_buffer[1:]
+            use_previous = False
+        else:
+            uncertain_pairs = deduper.uncertainPairs()
+            assert len(uncertain_pairs) == 1 
+            record_pair = uncertain_pairs[0]
+       
+        n_match_in_buffer = sum(len(x['match']) for x in examples_buffer)
+        n_distinct_in_buffer = sum(len(x['distinct']) for x in examples_buffer)
+                                                                                        
+        n_match, n_distinct = (len(deduper.training_pairs['match']) + n_match_in_buffer,
+                        len(deduper.training_pairs['distinct']) + n_distinct_in_buffer)
+      
         examples = {'distinct' : [], 'match' : []}
 
-        for record_pair in uncertain_pairs:
-            label = ''
-            labeled = False
-
-            for pair in record_pair:
-                for field in fields:
-                    line = "%s : %s" % (field, pair[field])
-                    print(line, file=sys.stderr)
-                print(file=sys.stderr) 
-
-            print("{0}/10 positive, {1}/10 negative".format(n_match, n_distinct),
-                  file=sys.stderr)
-            print('Do these records refer to the same thing?', file=sys.stderr)
-            valid_response = False
-            while not valid_response:
+        label = ''
+        labeled = False
+
+        for pair in record_pair:
+            for field in fields:
+                line = "%s : %s" % (field, pair[field])
+                print(line, file=sys.stderr)
+            print(file=sys.stderr) 
+
+        print("{0}/10 positive, {1}/10 negative".format(n_match, n_distinct),
+                file=sys.stderr)
+        print('Do these records refer to the same thing?', file=sys.stderr)
+        valid_response = False
+        while not valid_response:
+            if has_previous:
+                print('(y)es / (n)o / (u)nsure / (f)inished / (p)revious', file=sys.stderr)
+                label = input()
+                if label in ['y', 'n', 'u', 'f', 'p']:
+                    valid_response = True
+            else:
                 print('(y)es / (n)o / (u)nsure / (f)inished', file=sys.stderr)
                 label = input()
                 if label in ['y', 'n', 'u', 'f']:
                     valid_response = True
-
-            if label == 'y' :
-                examples['match'].append(record_pair)
-                labeled = True
-            elif label == 'n' :
-                examples['distinct'].append(record_pair)
-                labeled = True
-            elif label == 'f':
-                print('Finished labeling', file=sys.stderr)
-                finished = True
-            elif label != 'u':
-                print('Nonvalid response', file=sys.stderr)
-                raise
-
-        if labeled :
-            deduper.markPairs(examples)
+                if label == 'p':
```

Comment:
I put this for the user that would have seen the "previous" option once and who would not understand why the option would no longer be available. It could be removed though...